MovieQuiz/Presentation/AlertPresenter.swift

```diff
@@ -0,0 +1,32 @@
+//
+//  AlertPresenter.swift
+//  MovieQuiz
+//
+//  Created by Fixed on 17.05.2023.
+//
+
+import UIKit
+
+protocol AlertPresenterProtocol {
+    func showAlert(model: AlertModel)
+}
+
+class AlertPresenter: AlertPresenterProtocol {
+    weak private var delegate: UIViewController?
+    
+    init(delegate: UIViewController?) {
```

Comment:
учитывая что делегат у тебя приватное свойство, и его иначе чем через инициализатор не установить, то передавать сюда опционал неправильно:
```suggestion
    init(delegate: UIViewController) {
```
если передашь nil - то это будет весьма бесполезный презентер